# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLStyleEditor.cpp
# Commit: 712bef3646ff
# Full Hash: 712bef3646ffbe11627b35c95d84c654ad1d6cea
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-01-31 21:28:43
# Description:
#   Bug 1752668 - Restore null-check in `HTMLEditor::ClearStyleAt` r=m_kato
#   
#   I don't remember why I removed the null-check in
#   https://phabricator.services.mozilla.com/D131748#change-Q0aewmYc3vaK
#   
# ==============================================================================

diff -r cfee576f2b42 -r 712bef3646ff editor/libeditor/HTMLStyleEditor.cpp
--- a/editor/libeditor/HTMLStyleEditor.cpp	Mon Jan 31 17:13:49 2022 +0000
+++ b/editor/libeditor/HTMLStyleEditor.cpp	Mon Jan 31 17:15:35 2022 +0000
@@ -970,7 +970,8 @@
   // If it did split nodes, but topmost ancestor inline element is split
   // at start of it, we don't need the empty inline element.  Let's remove
   // it now.
-  if (HTMLEditUtils::IsEmptyNode(
+  if (splitResult.GetPreviousContent() &&
+      HTMLEditUtils::IsEmptyNode(
           *splitResult.GetPreviousContent(),
           {EmptyCheckOption::TreatSingleBRElementAsVisible,
            EmptyCheckOption::TreatListItemAsVisible,